src/Build/Instance/RunningObjectTable.cs

```diff
@@ -50,10 +60,27 @@ public object GetObject(string itemName)
             return obj;
         }
 
-        private IMoniker CreateMoniker(string itemName)
+        private Task<IMoniker> CreateMoniker(string itemName)
         {
-            Ole32.CreateItemMoniker("!", itemName, out IMoniker mk);
-            return mk;
+            // To avoid deadlock, create rot in a MTA thread. And the
+            // object will be MTA
+            var tcs = new TaskCompletionSource<IMoniker>();
+            Thread thread = new Thread(() =>
+            {
+                try
+                {
+                    Ole32.CreateItemMoniker("!", itemName, out IMoniker mk);
```

Comment:
Please help me check if I use task, thread correctly.
And CreateItemMoniker should also wrap in MTA thread right?

Reply:
>And CreateItemMoniker should also wrap in MTA thread right?

Yep. Good catch.